src/drivers/src/stm32f4xx.c

```
@@ -0,0 +1,87 @@
+#include "can_raw.h"
+
+#include "stm32f4xx_hal.h"
+#include <stdint.h>
+
+#define USING_CAN2
+
+static CAN_HandleTypeDef can1;
+static CAN_HandleTypeDef can2;
+static CAN_HandleTypeDef can3;
+
+static void config_handle_parameters(CAN_HandleTypeDef *hcan) {
+  // Universal configuration parameters
+  hcan->Init.Prescaler = 2;
+  hcan->Init.Mode = CAN_MODE_NORMAL;
+  hcan->Init.SyncJumpWidth = CAN_SJW_3TQ;
+  hcan->Init.TimeSeg1 = CAN_BS1_16TQ;
+  hcan->Init.TimeSeg2 = CAN_BS2_7TQ;
+  hcan->Init.TimeTriggeredMode = DISABLE;
+  hcan->Init.AutoBusOff = DISABLE;
+  hcan->Init.AutoWakeUp = DISABLE;
+  hcan->Init.AutoRetransmission = DISABLE;
+  hcan->Init.ReceiveFifoLocked = DISABLE;
+  hcan->Init.TransmitFifoPriority = DISABLE;
+}
+
+HAL_StatusTypeDef CANlib_Init(uint32_t baudrate) {
+  // TODO the baudrate argument should probably be removed...
```

You could return an error code if baudrate isn't 1M until this is fixed.